Recruit-Training/Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -54,7 +54,18 @@ bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) con
     QString timeString = sections.at(0);
     QTime time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    if(!time.isValid()){
+    if(!time.isValid())
+    {
+        return false;
+    }
+
+    if(sections.at(2).contains("[a-zA-Z]+"))
```

Comment:
I don't think this works as a regex, since it needs a QRegExp object passed in. Otherwise, it'll actually try to match the string `"[a-zA-Z]+"` :p

Take another look at how `toDouble()` works.